src/Compilers/Core/CodeAnalysisTest/Collections/SegmentedArrayHelperTests.cs

```diff
@@ -12,6 +17,89 @@ namespace Microsoft.CodeAnalysis.UnitTests.Collections
 {
     public class SegmentedArrayHelperTests
     {
+        [StructLayout(LayoutKind.Sequential, Size = 2)]
+        private struct Size2 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 4)]
+        private struct Size4 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 8)]
+        private struct Size8 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 12)]
+        private struct Size12 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        private struct Size16 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 24)]
+        private struct Size24 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 28)]
+        private struct Size28 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 32)]
+        private struct Size32 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 40)]
+        private struct Size40 { }
+
+        public static IEnumerable<object[]> ExplicitSizeTypes
```

Comment:
>object [](http://example.com/codeflow?start=34&length=6)

Can we use more specific element type?

Reply:
This is the signature for properties referenced by `[MemberData]`:
https://github.com/xunit/xunit/blob/2f51d3627a5df6bcdc9ca9b17764cac06a7d6488/src/xunit.core/MemberDataAttribute.cs#L13

Reply:
➡️ Left this code